worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -294,11 +297,7 @@ private[deploy] class Controller(
           },
           commitThreadPool)
 
-        if (future == null) {
-          future = task
-        } else {
-          future = CompletableFuture.allOf(future, task)
-        }
+        future = task
```

Comment:
why removed this CompletableFuture.allOf(future, task)?only one  task?

Reply:
Because future is guaranteed to be null according to the code.

Reply:
done